src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php

```diff
@@ -113,23 +113,18 @@ protected function createConfigurationDefinition()
     /**
      * @param Tokens $tokens
      *
-     * @return int[][] array of [start, end] indexes from sooner to later classes
+     * @return \Generator
      */
     private function findPhpUnitClasses(Tokens $tokens)
     {
         $phpUnitTestCaseIndicator = new PhpUnitTestCaseIndicator();
-        $phpunitClasses = [];
-
-        for ($index = 0, $limit = $tokens->count() - 1; $index < $limit; ++$index) {
```

Comment:
in original solution, we were iterating each element of collection (having 1k+ tokens usually!) only once - `O(n)`.
now, its `O(2n)`.
In fact, for non-phpunit it will iterate only once as well, yet for phpunit, it will literally iterate almost all of the element twice, as usually test file has just one big test class.

Reply:
this could be fixed by keeping tracking of the close brace of the class, removing much of the overhead;
will your analyze is correct it is also incomplete, the new code provides two benefits;
- not returning an array after _all_ tokens are checked, the gain here is up to the caller
- no need for array flipping the results for some fixers which caused the returned array to be copied

Reply:
most cases:
- iterate over all classes
- flip one-item array

performance loss here is significant